luizfiorentino/my-city-app
# Status update feat

# What this pull request does

## What

Creates a component "EditBar" that appears either in the upper part or in the footer of the DetailsPage.
There are three buttons "Edit", "Mark as solved" and "Delete" with the respective actions.
A modal opens with a confirmation message and a selector (only in the "Edit" case) for changing the status if necessary appears, and a "Cancel" or "Confirm" button.

## Why

This feature allows the admin to update an existing issue (selecting a new status if needed and writing a message summarising the update), marking the issue as "solved" (in which case there's no need to write a message which is default), or delete the issue, including the related updates.

## How

There is a function "updateStatus" in the EditBar component that passes as parameters "message", "status", and "buttonMode" to another function, either "sendUpdateIssueRequest", "sendDeleteRequest", or "sendSolvedUpdateRequest", depending on the button mode (edit, mark as solved or delete). These functions do a POST request (Edit/Solved) or DELETE request to the api handlers via the "/api/statusChanges" or "/api/issues/${issueId}" routes, so the database will be be accordingly updated.
The EditBar has useState for the message the user types in, the selected status and the buttonMode (edit, mark as solved, delete), the latter determined which function will be called.

## Steps to test this

On the DetailsPage, check out the EditBar on the footer (mobile version), or the top (from tablet screen on) with the three buttons: "Edit", "Mark as solved", and "Delete". Click each button and see a message to confirm the operation, or to type a message and select a different status (only Edit case). 
It should display the new status immediately or redirect to the aminList page in case of delete (without the chosen issue).

## Anything else

This is a relatively complex feature that could have been done in smaller chunks. By doing as we did it, it has the advantage to address the feature in a systematic way.
